pkg/solana/client/multinode/transaction_sender.go

```diff
@@ -94,13 +91,16 @@ type TransactionSender[TX any, CHAIN_ID ID, RPC SendTxRPCClient[TX]] struct {
 // * If there is at least one terminal error - returns terminal error
 // * If there is both success and terminal error - returns success and reports invariant violation
 // * Otherwise, returns any (effectively random) of the errors.
-func (txSender *TransactionSender[TX, CHAIN_ID, RPC]) SendTransaction(ctx context.Context, tx TX) (SendTxReturnCode, error) {
-	txResults := make(chan sendTxResult)
-	txResultsToReport := make(chan sendTxResult)
+func (txSender *TransactionSender[TX, RESULT, CHAIN_ID, RPC]) SendTransaction(ctx context.Context, tx TX) RESULT {
+	txResults := make(chan RESULT)
+	txResultsToReport := make(chan RESULT)
 	primaryNodeWg := sync.WaitGroup{}
 
-	ctx, cancel := txSender.chStop.Ctx(ctx)
-	defer cancel()
+	if txSender.State() != "Started" {
+		return txSender.newResult(errors.New("TransactionSender not started"))
+	}
+
+	ctx, _ = txSender.chStop.Ctx(ctx)
```

Comment:
This seems like a bug in the initial version of the txSender, so feel free to create it as a separate PR (but we'll have to merge it first)
We should only pass caller's context into `txSender.collectTxResults`. All other calls are background and can not be canceled by the caller, so we should pass txSender's root context.

Reply:
Close funcs must not be ignored because they leak goroutines/memory:
```suggestion
	ctx, close = txSender.chStop.Ctx(ctx)
	defer close()
```